app/test/shared/test_services.dart

```diff
@@ -59,7 +59,7 @@ void testWithServices(
       await fork(() async {
         registerAuthProvider(FakeAuthProvider());
         registerDomainVerifier(FakeDomainVerifier());
-        registerEmailSender(FakeEmailSender());
+        registerEmailSender(fakeEmailSender.sendMessage);
```

Comment:
This should be a new instance each time this block is running (and the reason of test failures), otherwise we'd need to clear the buffered list of emails before checking them.

Reply:
Ah, then sticking to a class with a single method is perhaps better..